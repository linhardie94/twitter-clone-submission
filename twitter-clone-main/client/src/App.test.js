// import { render, screen } from '@testing-library/react';
// import App from './App';

// test('renders learn react link', () => {
//   render(<App />);
//   const linkElement = screen.getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });

import React from 'react';
import { render, screen } from '@testing-library/react';
import RightSidebar from '../src/components/RightSidebar/RightSidebar';

//snapshot test
test('RightSidebar component should match snapshot', () => {
  const { container } = render(<RightSidebar />);
  expect(container.firstChild).toMatchSnapshot();
});

//snapshot test
test('RightSidebar component should match snapshot', () => {
  const { getByText } = render(<RightSidebar />);
  expect(getByText('Trending')).toHaveClass('font-medium');
  expect(getByText('#gryffindor')).toHaveClass('font-bold');
  //test for other classes
});

//unit test
test('RightSidebar should render with correct text', () => {
  const { getByText } = render(<RightSidebar />);
  expect(getByText('Trending')).toBeInTheDocument();
  expect(getByText('#gryffindor')).toBeInTheDocument();
  expect(getByText('#hufflepuff')).toBeInTheDocument();
  expect(getByText('#slytherin')).toBeInTheDocument();
  expect(getByText('#ravenclaw')).toBeInTheDocument();
});