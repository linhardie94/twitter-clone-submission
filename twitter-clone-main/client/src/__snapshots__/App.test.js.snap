// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RightSidebar component should match snapshot 1`] = `
<div
  class="p-6 bg-slate-100 rounded-lg mx-4 space-y-4"
>
  <h2
    class="font-medium"
  >
    Trending
  </h2>
  <p
    class="font-bold"
  >
    #gryffindor
  </p>
  <p
    class="font-bold"
  >
    #hufflepuff
  </p>
  <p
    class="font-bold"
  >
    #slytherin
  </p>
  <p
    class="font-bold"
  >
    #ravenclaw
  </p>
</div>
`;
